sdk/synapse/Azure.Analytics.Synapse.AccessControl/src/SynapseAccessControlClient.cs

```diff
@@ -54,6 +55,497 @@ public virtual Response<SynapseRoleAssignment> CreateRoleAssignment(SynapseRoleS
+                        }),
+                    new RequestOptions()
+                    {
+                        CancellationToken = cancellationToken
+                    }).ConfigureAwait(false);
+
+                JsonDocument roleAssignment = await JsonDocument.ParseAsync(createRoleAssignmentResponse.ContentStream, default, cancellationToken).ConfigureAwait(false);
+
+                return Response.FromValue(new SynapseRoleAssignment(
+                    roleAssignment.RootElement.GetProperty("id").ToString(),
+                    new SynapseRoleAssignmentProperties(
+                        roleAssignment.RootElement.GetProperty("principalId").ToString(),
+                        roleAssignment.RootElement.GetProperty("roleDefinitionId").ToString(),
+                        roleAssignment.RootElement.GetProperty("scope").ToString())),
+                    createRoleAssignmentResponse);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        public virtual Response DeleteRoleAssignment(SynapseRoleScope roleScope, string roleAssignmentName, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNullOrEmpty(roleAssignmentName, nameof(roleAssignmentName));
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(SynapseAccessControlClient)}.{nameof(DeleteRoleAssignment)}");
+            scope.Start();
+            try
+            {
+                return DeleteRoleAssignmentById(
+                    roleAssignmentName,
+                    roleScope.ToString(),
+                    new RequestOptions()
+                    {
+                        CancellationToken = cancellationToken
+                    });
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        public virtual async Task<Response> DeleteRoleAssignmentAsync(SynapseRoleScope roleScope, string roleAssignmentName, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNullOrEmpty(roleAssignmentName, nameof(roleAssignmentName));
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(SynapseAccessControlClient)}.{nameof(DeleteRoleAssignment)}");
+            scope.Start();
+            try
+            {
+                return await DeleteRoleAssignmentByIdAsync(
+                    roleAssignmentName,
+                    roleScope.ToString(),
+                    new RequestOptions()
+                    {
+                        CancellationToken = cancellationToken
+                    }).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        // TODO: do not use roleScope ?
```

Comment:
It will be interesting to understand why the ARM and KV RBAC APIs have this and Synapse API doesn't.  I agree that we shouldn't take roleScope as a parameter if we're not using it in the method.